tests/test_hyperbahn.py

```diff
@@ -0,0 +1,120 @@
+# Copyright (c) 2018 Uber Technologies, Inc.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+from __future__ import absolute_import
+
+import pytest
+from tornado import gen
+
+from tchannel import TChannel
+
+
+@pytest.yield_fixture
+def server(io_loop):
+    server = TChannel('server')
+
+    @server.json.register('hello')
+    def hello(request):
+        return 'world'
+
+    server.listen()
+    try:
+        yield server
+    finally:
+        server.close()
+
+
+@pytest.yield_fixture
+def client(io_loop):
+    client = TChannel('client')
+    client.listen()
+    try:
+        yield client
+    finally:
+        client.close()
+
+
+@pytest.yield_fixture
+def routers(server, io_loop):
+    routers = [TChannel('hyperbahn') for i in range(5)]
```

Comment:
Uh hello, `xrange` for mo' betta efficiency. Gotta count those nanos.

Reply:
wontfix. this breaks python 3 compatibility.